extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -0,0 +1,152 @@
+		if (!locations) {
+			return;
+		}
+		// publishSettingsFile = locations[0];
+	}
+
+	// 2. Select connection
+	// TODO: Hook up to MSSQL
+	const connectionProfile = await vscode.window.showQuickPick(
+		['Connection 1', 'Connection 2', 'Create New Connection'],
+		{ title: constants.selectConnection, ignoreFocusOut: true });
+	if (!connectionProfile) {
+		return;
+	}
+	const dbs = ['db1', 'db2'];
+	const dbQuickpicks = dbs.map(db => {
+		return {
+			label: db,
+			dbName: db
+		} as vscode.QuickPickItem & { dbName: string, isCreateNew?: boolean };
+	});
+	// Ensure the project name is an option, either adding it if it doesn't already exist or moving it to the top if it does
+	const projectNameIndex = dbs.findIndex(db => db === project.projectFileName);
+	if (projectNameIndex === -1) {
+		dbQuickpicks.unshift({ label: constants.newDatabaseTitle(project.projectFileName), dbName: project.projectFileName });
+	} else {
+		dbQuickpicks.splice(projectNameIndex, 1);
+		dbQuickpicks.unshift({ label: project.projectFileName, dbName: project.projectFileName });
+	}
+
+	dbQuickpicks.push({ label: constants.createNew, dbName: '', isCreateNew: true });
+	// 3. Select database
+	// TODO: Hook up to MSSQL
+	let databaseName = '';
+	while (databaseName === '') {
+		const selectedDatabase = await vscode.window.showQuickPick(
+			dbQuickpicks,
+			{ title: constants.selectDatabase, ignoreFocusOut: true });
+		if (!selectedDatabase) {
+			// User cancelled
+			return;
+		}
+		databaseName = selectedDatabase.dbName;
+		if (selectedDatabase.isCreateNew) {
+			databaseName = await vscode.window.showInputBox(
+				{
+					title: constants.enterNewDatabaseName,
+					ignoreFocusOut: true
+				}
+			) ?? '';
+			// If user cancels out of this just return them to the db select quickpick in case they changed their mind
+		}
+	}
+
+
+	// 4. Modify sqlcmd vars
+	// TODO: Concat ones from publish profile
+	let sqlCmdVariables = Object.assign({}, project.sqlCmdVariables);
+
+	if (Object.keys(sqlCmdVariables).length > 0) {
+		// Loop over the vars to modify until the user is done
+		while (true) {
+			const quickPickItems = Object.keys(sqlCmdVariables).map(key => {
+				return {
+					label: key,
+					description: sqlCmdVariables[key],
+					key: key
+				} as vscode.QuickPickItem & { key?: string, isResetAllVars?: boolean, isDone?: boolean };
+			});
+			quickPickItems.push({ label: constants.resetAllVars, isResetAllVars: true });
+			quickPickItems.unshift({ label: constants.done, isDone: true });
+			const sqlCmd = await vscode.window.showQuickPick(
+				quickPickItems,
+				{ title: constants.chooseSqlcmdVarsToModify, ignoreFocusOut: true }
+			);
+			if (!sqlCmd) {
+				// When user hits escape then we continue on here, we don't exit the publish
+				// flow since this is an optional step
+				break;
+			}
+			if (sqlCmd.key) {
+				const newValue = await vscode.window.showInputBox(
+					{
+						title: sqlCmd.key,
```

Comment:
might be helpful if the title had more text like "Enter value for 'sqlCmd.key'"

Reply:
![image](https://user-images.githubusercontent.com/28519865/124681791-28ff5600-de7e-11eb-8227-024d547b4980.png)